### PR TITLE
Implement ruler proxy

### LINKED ISSUE #19: Support more components
In the config file, we currently support the distributor and the query-frontend components. As @songjiayang suggested in [this issue](https://github.com/cortexproject/auth-gateway/issues/10#issuecomment-1519621015), we should support alertmanager and ruler components as well. After #6 is merged, the necessary modifications need to be made in the config file.

### PR DESCRIPTION
**Pull request by dogukanteber**
This PR implements ruler proxy. It also partly resolves #19.

### CODE REVIEW

File: gateway/gateway.go [hunk 1]
```diff
@@ -45,6 +46,16 @@ var defaultAlertmanagerAPIs = []string{
 	"/multitenant_alertmanager/delete_tenant_config",
 }
 
+var defaultRulerAPIs = []string{
+	"/prometheus/api/v1/rules",
+	"/api/prom/api/v1/rules",
+	"/prometheus/api/v1/alerts",
+	"/api/prom/api/v1/alerts",
+	"/api/v1/rules/",
```
Comment (friedrichg) [inline gateway/gateway.go:54]:
```suggestion
	"/api/v1/rules",
	"/api/v1/rules/",
```
it's also needed

File: gateway/proxy.go [hunk 2]
```diff
@@ -35,6 +36,12 @@ var defaultTimeoutValues map[string]Upstream = map[string]Upstream{
 		HTTPClientTLSHandshakeTimeout:   time.Second * 5,
 		HTTPClientResponseHeaderTimeout: time.Second * 5,
 	},
+	RULER: {
+		HTTPClientTimeout:               time.Second * 10,
+		HTTPClientDialerTimeout:         time.Second * 3,
+		HTTPClientTLSHandshakeTimeout:   time.Second * 4,
+		HTTPClientResponseHeaderTimeout: time.Second * 3,
```
Comment (friedrichg) [inline gateway/proxy.go:43]:
```suggestion
		HTTPClientTimeout:               time.Second * 15,
		HTTPClientDialerTimeout:         time.Second * 5,
		HTTPClientTLSHandshakeTimeout:   time.Second * 5,
		HTTPClientResponseHeaderTimeout: time.Second * 5,
```
No reason, for now, to have a different timeout from the others.

**Review by friedrichg — CHANGES_REQUESTED** [review]
minor nits only. Looks good

File: gateway/proxy.go (hunk 2 above)
Comment (dogukanteber) [reply gateway/proxy.go:43]: when we talked on slack, we decided to have 10 seconds for the ruler's HTTPClientTimeout.

Comment (friedrichg) [reply gateway/proxy.go:43]: did we?. I can't remember now 😅 . Can you remind me a reason for that?

**Review by friedrichg — APPROVED** [review]

**Pull request merged**